src/ui/public/field_editor/components/scripting_help/test_script.js

```diff
@@ -0,0 +1,226 @@
+    this.setState({
+      additionalFields: selectedOptions
+    });
+  }
+
+  renderPreview() {
+    const { previewData } = this.state;
+
+    if (!previewData) {
+      return null;
+    }
+
+    if (previewData.error) {
+      return (
+        <EuiCallOut
+          title="There's an error in your script"
+          color="danger"
+          iconType="cross"
+        >
+          <EuiCodeBlock
+            language="json"
+            className="scriptPreviewCodeBlock"
+            data-test-subj="scriptedFieldPreview"
+          >
+            {JSON.stringify(previewData.error, null, ' ')}
+          </EuiCodeBlock>
+        </EuiCallOut>
+      );
+    }
+
+    return (
+      <Fragment>
+        <EuiTitle size="xs"><p>First 10 results</p></EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiCodeBlock
+          language="json"
+          className="scriptPreviewCodeBlock"
+          data-test-subj="scriptedFieldPreview"
+        >
+          {JSON.stringify(previewData, null, ' ')}
+        </EuiCodeBlock>
+      </Fragment>
+    );
+  }
+
+  renderToolbar() {
+    const fieldsByTypeMap = new Map();
+    const fields = [];
+
+    this.props.indexPattern.fields
+      .filter(field => {
+        return !field.name.startsWith('_');
+      })
+      .forEach(field => {
+        if (fieldsByTypeMap.has(field.type)) {
+          const fieldsList = fieldsByTypeMap.get(field.type);
+          fieldsList.push(field.name);
+          fieldsByTypeMap.set(field.type, fieldsList);
+        } else {
+          fieldsByTypeMap.set(field.type, [field.name]);
+        }
+      });
+
+    fieldsByTypeMap.forEach((fieldsList, fieldType) => {
+      fields.push({
+        label: fieldType,
+        options: fieldsList.sort().map(fieldName => {
+          return { value: fieldName, label: fieldName };
+        })
+      });
+    });
+
+    fields.sort((a, b) => {
+      if (a.label < b.label) return -1;
+      if (a.label > b.label) return 1;
+      return 0;
+    });
+
+    return (
+      <Fragment>
+        <EuiFormRow
+          label="Additional fields"
+        >
+          <EuiComboBox
+            placeholder="Select..."
+            options={fields}
+            selectedOptions={this.state.additionalFields}
+            onChange={this.onAdditionalFieldsChange}
+            data-test-subj="additionalFieldsSelect"
+          />
+        </EuiFormRow>
+
+        <EuiButton
+          onClick={this.previewScript}
+          disabled={this.props.script ? false : true}
+          isLoading={this.state.isLoading}
+          data-test-subj="runScriptButton"
+        >
+          Run script
+        </EuiButton>
+      </Fragment>
+    );
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <EuiSpacer />
+        <EuiText>
+          <h3>Preview results</h3>
+          <p>
+            Run your script to preview the first 10 results. You can also select some
+            additional fields to include in your results to gain more context.
```

Comment:
would be nice to clarify this a little bit. initially, I thought that adding additional fields somehow affected my actual script, but they are only shown as part of the preview.

not sure if there's value, but maybe we can also add a "Show query" link that displays the request, which will include the additional `_source` fields

Reply:
How about this:
```
Run your script to preview the first 10 results. You can also select some
additional fields to include in each result to help tell your results apart.
```